docs/tutorials/gpu_setup.rst

```diff
@@ -0,0 +1,176 @@
+GPU Acceleration
+----------------
+
+Installation and Configuration
+==============================
+
+Installing CUDA Toolkit
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The first step in utilizing the computational power of your Nvidia GPU
+is to install the CUDA toolkit. (If you've already configured your GPU for other software, you may skip this step.) To download the installer, visit `this
+link <https://developer.nvidia.com/cuda-downloads?target_os=Windows&target_arch=x86_64&target_version=10&target_type=exe_network>`__
+and provide your system info to download the installer. You must be
+using either Linux or Windows, and you must be using one of
+`these <https://developer.nvidia.com/cuda-gpus>`__ graphics cards. Once
+the toolkit has been installed, follow the instructions in the installer
+GUI. Once complete, restart your computer.
+
+Python and PyTorch GPU configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The next step is to check whether your Python and PyTorch installations are correctly configured to use your GPU. After installing MPoL, you can check whether everything installed correctly by opening up a Python interpreter, ard running
+
+.. code:: ipython3
+
+    import torch
+    print(torch.cuda.is_available())
+
+This command should return ``True``. If not, then you may need to use a more specific installation process. Go to the `PyTorch Official Site <https://pytorch.org/>`__ and scroll down
+on the page until you see the **Install PyTorch** section. Input your
+specifications for your needs into this area and use the text that is
+generated for your install. For example, making of this tutorial on a Windows
+10 system with a Nvidia GTX 1080 required specific pip installation,
+while another Windows 10 system using a Nvidia GTX 1660Ti worked with the default
+``pip install torch torchvision``. Your mileage may vary.
+
+Why use the GPU?
+================
+
+Using a GPU can accelerate computing speeds up to 100x over CPUs, especially for operations on large images, like is common for MPoL. The following is a quick example showing the addition of two large vectors. Your exact timing may vary, but for our hardware this calculation took
+133 milliseconds seconds on the CPU, while it only took 35 milliseconds on the GPU.
+
+.. code:: ipython3
+
+    import torch
+    import time
+    N = int(9.9e7)
+    A = torch.ones(N)
+    B = torch.ones(N)
+    start = time.time()
+    C = A + B
+    print(time.time() - start)
+
+.. code:: ipython3
+
+    torch.cuda.empty_cache() # emptying the cache on the gpu just incase there was any memory left over from an old operation
+    A = A.cuda()
+    B = B.cuda()
+
+.. code:: ipython3
+
+    start = time.time()
+    C = A + B
+    print(time.time() - start)
```

Comment:
Is this timing correct if you view the calculation on the GPU? Sometimes GPU events execute asynchronously, so you might not actually be timing the correct thing here (e.g., https://discuss.pytorch.org/t/how-to-measure-time-in-pytorch/26964). It could be correct, I just wanted to make sure.

Reply:
Ah you were right, the timing that I was reporting was approx. 100x longer than the time reported by torch. I will update this now.